src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -154,18 +155,26 @@ def run(self, inputs: List[numpy.ndarray], val_inp: bool) -> List[numpy.ndarray]
         If the self.internal_cache_active=True, the internal
         deepsparse kv cache management is enabled. In this case
         the LIB.kv_cache class object will be passed to the engine
-        call as well.
+        call as well. In this scenario also the inputs will not be
+        validated, even if the val_inp=True. This is because we
+        want to pass the empty kv cache inputs (batch_size=0) to
+        the engine.
 
         :param inputs: The inputs to run the engine with
         :param val_inp: Whether the input is for validation or not
         :return: The output of the engine
         """
 
         if self.internal_cache_active:
-            # validate the inputs if needed
             if val_inp:
-                self.engine._validate_inputs(inputs)
-            # run the engine with the LIB.kv_cache object
+                warnings.warn(
+                    "Internal kv cache is active. "
+                    "To maximize inference efficiency, "
+                    "the engine inputs pertaining to "
+                    "the kv cache values are empty arrays. "
+                    "This is why, even though val_inp=True,"
+                    "the inputs will not be validated"
+                )
```

Comment:
is it set to true by default? because this is an internally managed input i don't see the reason to check the input by default. also, will this be printed for every internal inference i.e. `run()` call?

Reply:
just discussed it with Ben. We will remove this check (if-statement), as well as the warning, and just leave a comment about why we are refraining from validating the inputs. Thanks @mgoin